projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js

```diff
@@ -19,21 +19,30 @@ import PublicizeConnectionVerify from '../connection-verify';
 import PublicizeForm from '../form';
 import PublicizeTwitterOptions from '../twitter/options';
 import useSelectSocialMediaConnections from '../../hooks/use-social-media-connections';
-import usePostJustSave from '../../hooks/use-post-just-saved';
+import { usePostJustBeforeToPublish } from '../../hooks/use-saving-post';
 
 const PublicizePanel = ( { prePublish } ) => {
-	const { refresh, hasEnabledConnections } = useSelectSocialMediaConnections();
+	const { refresh, connections } = useSelectSocialMediaConnections();
 
 	// Refresh connections when the post is just saved.
-	usePostJustSave(
+	usePostJustBeforeToPublish(
 		function () {
-			if ( ! hasEnabledConnections ) {
-				return;
-			}
-
-			refresh();
+			/*
+			 * Being optimistic, it sets the connections
+			 * that are going to be used
+			 * to share the post as `done`.
+			 * The sharing process is handled by an async action,
+			 * in the server-side.
+			 */
+			const updatedConnections = connections.map( connection => ( {
+				...connection,
+				done: connection.enabled,
+				toggleable: ! connection.enabled,
+			} ) );
+
+			refresh( updatedConnections );
```

Comment:
When does `done` get updated on the server? Could this refresh cause `done` to be updated to `false` again, and if not could we just disable the UI while the refresh completes?

Reply:
Also are we essentially setting the connections to the updated values? If so, I wonder if this should be a separate update action, and we can leave `refresh` to make the API call.

Reply:
> When does done get updated on the server?

I wouldn't say updated but assigned. It should be done by a sync job. The state is store in the _wpas_done_ meta if I'm not wrong.

> Could this refresh cause done to be updated to false again...

I'm not sure. I think it's possible cause these flags are handled by the async job. Anyway, we should block re-sharing the same connection when there is a current sharing in transit.

> , and if not could we just disable the UI while the refresh completes?

We disable the UI setting toggle to false, too.

Reply:
> Also are we essentially setting the connections to the updated values?  If so, I wonder if this should be a separate update action, and we can leave refresh to make the API call.

That's a good sugestion. what name do you suggest? `update()`?